### ISSUE TITLE
False positives in `modernize-use-default-member-init` in clang-tidy 21

### ISSUE THREAD

**Issue by arvid-norlander** [llvm/llvm-project#156295]
The release notes for [clang-tidy 21](https://releases.llvm.org/21.1.0/tools/clang/tools/extra/docs/ReleaseNotes.html#improvements-to-clang-tidy) reads:
> Improved [modernize-use-default-member-init](https://releases.llvm.org/21.1.0/tools/clang/tools/extra/docs/clang-tidy/checks/modernize/use-default-member-init.html) check by matching arithmetic operations, constexpr and static values, and detecting explicit casting of built-in types within member list initialization.

However there seem to be some false positives when there is an expression that partly depends on a constructor argument. E.g.:

```c++
#include <cstdint>

constexpr double CONSTANT = 2.0;

class Example1
{
   explicit Example1(const int32_t param)
   : member(
        static_cast<int32_t>(CONSTANT / (static_cast<double>(param) * 0.001)))
   {
   }

   const int32_t member;
};
```

This results in:

```
test.cpp:13:18: error: use default member initializer for 'member' [modernize-use-default-member-init,-warnings-as-errors]
    8 |    : member(
      |      ~~~~~~~
    9 |         static_cast<int32_t>(CONSTANT / (static_cast<double>(param) * 0.001)))
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   10 |    {
   11 |    }
   12 | 
   13 |    const int32_t member;
      |                  ^     
```

Using `-fix` creates the invalid code below:

```c++
#include <cstdint>

constexpr double CONSTANT = 2.0;

class Example1
{
   explicit Example1(const int32_t param)
    
   {
   }

   const int32_t member{
        static_cast<int32_t>(CONSTANT / (static_cast<double>(param) * 0.001))};
};
```

**Comment by vbvictor** [comment]
CC @RiverDave
